### DISCUSSION TITLE
Bandpass filter

### DISCUSSION THREAD

**Discussion by Navids94** [geodynamics/sw4#201] [Q&A]
Hi, I just wondering if I can use bandpass prefilter? since when I use bandpass I get following error:
prefilter fc1=0.001 fc2=10 type=bandpass passes=2 order=2
WARNING: the 2 pass prefilter has an estimated precursor of length 2.700949e+03 s
*** To avoid artifacts due to sudden startup, increase t0 in the source named 'Liu' by at least 2.699949e+03

**Comment by andersp** [comment]
You can use a band-pass filter in the prefilter command, but since sw4 applies the filter both forwards and backwards in time, you need to make t0 in the source function large enough to avoid a sudden startup of the prefiltered source time function. The problem with the high-pass part of a band-pass filter is that the frequency is very low, corresponding to a very slow decay of the tail of the filter. It is more practical and computationally efficient to apply the high pass part of the filter to the time series after the simulations are done, in a post processing step.

**Reply by Navids94** [reply]
Thank you very much for your response. For prefiltering, I will use the lowpass(fc2=5.0 for resolution of 5Hz) command, and then the high pass after simulations.
